firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -167,16 +167,21 @@ public FieldPath inequalityField() {
     return null;
   }
 
-  public boolean hasArrayContainsFilter() {
+  @Nullable
+  /**
+   * Checks if any of the provided RelationOps are included in the query and returns the first one
+   * that is, or null if none are.
+   */
+  public Operator findOperatorFilter(List<Operator> filterOps) {
```

Comment:
optional: Looking at this code today, "findOperatorFilter()" doesn't seem like the clearest name.  And it's unfortunate that TypeScript and Android end up with different names.  I would be slightly in favor of renaming both to "findFilterOperator()"

Reply:
Good idea. I'll leave this comment unresolved until this PR is close to mergeable and I'll update the TS sdk.

Reply:
Gil will probably update this when he ports over https://github.com/firebase/firebase-js-sdk/pull/1894/files